### PR TITLE
Fix self deadlock and some compiler/linker error exposed bu GCC8

### PR DESCRIPTION
**Pull request by munroesj52**
This is the fix for: Self deadlock issue in single threaded apps #44

### CODE REVIEW

File: src/sascompoundheap.cpp [hunk 1]
```diff
@@ -1669,3 +1673,4 @@ SPHCompoundPCQNearAlloc (void *nearObj)
     }
   return newHeap;
 }
+
```
Comment (RajalakshmiS) [inline src/sascompoundheap.cpp]: The changes look good in general except few extra  new empty lines added. The commit message did not appear. Could you please add that as a comment which I will use while merging the changes.

Comment (munroesj52) [reply src/sascompoundheap.cpp]: [git-log-20180115.txt](https://github.com/sphde/sphde/files/1636614/git-log-20180115.txt)
[git-log-20180115b.txt](https://github.com/sphde/sphde/files/1636615/git-log-20180115b.txt)

Comment (munroesj52) [reply src/sascompoundheap.cpp]: Looks like I have core.editor set:

sjmunroe@SJM-ThinkPad-W520:~/work/github/forks/sphde$ git config --global core.editor
gedit
sjmunroe@SJM-ThinkPad-W520:~/work/github/forks/sphde$ git config core.editor
gedit

**Comment by RajalakshmiS** [comment]
Merged.

**Pull request closed without merging**